src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -628,11 +617,13 @@ void ov::npuw::JustInferRequest::bind_global_parameters(std::size_t idx) {
 
     // Run host-side gather, if required
     if (comp_model_desc.host_gather.dst_idx != -1) {
-        auto& dst = comp_model_desc.closure[comp_model_desc.host_gather.dst_idx - comp_model_desc.param_base];
+        const auto& gport = comp_model_desc.compiled_model->inputs()[comp_model_desc.host_gather.dst_idx];
+        const auto gather = subr->get_tensor(gport);
+
         const auto& vocab = comp_model_desc.closure[comp_model_desc.host_gather.src_idx - comp_model_desc.param_base];
         const auto& lport = comp_model_desc.compiled_model->inputs()[comp_model_desc.host_gather.idx_idx];
         const auto lookup = subr->get_tensor(lport);
-        ov::npuw::util::gather(ov::get_tensor_impl(vocab), lookup, ov::get_tensor_impl(dst));
+        ov::npuw::util::gather(ov::get_tensor_impl(vocab), lookup, gather);
```

Comment:
@smirnov-alexey THIS is a problem that we've missed. Finally I've found that :)

The problem was that with this branch, the target model worked but all others stopped.
I've found that in a model I was enabling there's no host gather - pattern didn't match - gather was done by the subgraph model itself. But in all other networks which broke, it was there.

On CPU, all was working fine. On NPU though, the output was empty.

I've found that my host gather implementation is silly: as it registers an extra parameter, it also has to register a new closure. When host_gather runs, we gather embeddings from a vocabulary in a closure - at the early beginning of the subgraph's execution, at `bind_global_parameters`. 

Previously (before this PR), once host gather ran, we also did `unpack_closure` where that dummy host-side closure was copied-or-set into the respective input tensor.

With this PR, it is done only once in the constructor. So if it is CPU, we pick "set" path and set the closure's gather target tensor into the request's input tensor - and it worked.

On NPU though we selected the "copy" path in the closure, so.. we copied a dull tensor into the input tensor and that's it. When host_gather ran during the inference, it updated the dull tensor but not the input's tensor as the graph as already "pre-set" with its closure.

So I've spent a week thinking of what went wrong here. :)